core/domain/user_jobs_continuous.py

```diff
@@ -380,6 +380,7 @@ def _handle_incoming_event(cls, active_realtime_layer, event_type, *args):
         """
         exp_id = args[0]
 
+        @transaction_services.run_in_transaction_wrapper
         def _refresh_average_ratings(user_id, rating, old_rating):
```

Comment:
My Thanks @lkbhitesh07! only suggestion would be to change the name to _refresh_average_ratings_transactional() (and similarly below) so that developers can see what's going on at the callsite (and reason accordingly).

I'll amend the original issue accordingly too. /cc @vojtechjelinek

Reply:
Done @seanlip PTAL